### PR TITLE
Book controller

### PR DESCRIPTION
**Pull request by vfilimo**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -90,6 +103,34 @@
...
 					<linkXRef>false</linkXRef>
+					<sourceDirectories>src</sourceDirectories>
+				</configuration>
+			</plugin>
+
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-compiler-plugin</artifactId>
+				<configuration>
+					<source>${java.version}</source> <!-- depending on your project -->
```
Comment (Elena-Bruyako) [inline pom.xml]: remove comments

File: src/main/java/online/store/book/controller/BookController.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import online.store.book.service.BookService;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping(value = "/books")
```
Comment (Elena-Bruyako) [inline src/main/java/online/store/book/controller/BookController.java]:
```suggestion
@RequestMapping("/books")
```

File: src/main/java/online/store/book/repository/BookRepositoryImpl.java [hunk 3]
```diff
@@ -47,4 +48,15 @@ public List<Book> getAll() {
...
     }
+
+    @Override
+    public Optional<Book> findBookById(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            Query<Book> query = session.createQuery("FROM Book b WHERE b.id = :id", Book.class);
+            query.setParameter("id", id);
+            return Optional.ofNullable(query.getSingleResultOrNull());
+        } catch (HibernateException e) {
+            throw new DataProcessingException("Can't find book with id: " + id + " in the DB", e);
```
Comment (Elena-Bruyako) [inline src/main/java/online/store/book/repository/BookRepositoryImpl.java]:
```suggestion
             return Optional.ofNullable(session.get(Book.class, id));
        } catch (Exception e) {
            throw new DataProcessingException("Can't find book with id: " + id + " in the DB", e);
        }
```

File: src/main/java/online/store/book/service/BookServiceImpl.java [hunk 4]
```diff
@@ -10,14 +15,28 @@
...
     }
 
     @Override
-    public List<Book> getAll() {
-        return bookRepository.getAll();
+    public List<BookDto> getAll() {
+        return bookRepository.getAll()
+                .stream()
+                .map(bookMapper::toDto)
+                .collect(Collectors.toList());
```
Comment (Elena-Bruyako) [inline src/main/java/online/store/book/service/BookServiceImpl.java]:
```suggestion
                .toList();
```

File: src/main/java/online/store/book/service/BookServiceImpl.java [hunk 5]
```diff
@@ -10,14 +15,28 @@
...
+
```
Comment (Elena-Bruyako) [inline src/main/java/online/store/book/service/BookServiceImpl.java]:
```suggestion
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

**Review by okuzan — APPROVED** [review]

**Pull request merged**